docs/resources/client/i18n.md

```diff
@@ -120,6 +120,31 @@ Some places in Minecraft offer you helper methods to get a translation keys. For
 The only purpose of translation keys is for localization. Do not use them for game logic, that's what [registry names][regname] are for.
 :::
 
+### Component Translations
+
+NeoForge extends the vanilla language file format to allow using arrays of `Component`s in translations.
+Additionally, NeoForge registers the inserting component (type `neoforge:inserting`), which inserts
+an argument from the surrounding translation.
+
+For example, you can modify the Vanilla chat format like so:
+```json5
+{
+  "chat.type.text": [
+    // The empty string is needed as (by Vanilla's JSON component format)
+    // the first element in an array is the parent of all others.
+    "",
+    {"text": "A mysterious stranger known only as <", "color": "gray"},
+    {"index": 0, "obfuscated": true},
```

Comment:
This is also not clear that it refers to the arguments provided by the translation, especially since those were 1-indexed.